### PR TITLE
(RPW1-18): freewheel - Add skip option

### PR DESCRIPTION
**Pull request by psaint-j**
# About the type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Test plan

- [ ] Empty data
- [ ] Invalid data
- [ ] Correct data

# Checklist
### Before code review

- [x] Move Jira ticket into PR
- [x] Check with another developers that features sounds good

### After code review

- [ ] Move Jira ticket into QA

### After QA validation
If conflicts are present in github:
  - [ ] `git pull --rebase origin master`
  - [ ] Fix conflicts
  - [ ] `git rebase --continue` repeat until there is no more conflicts
  - [ ] `git push --force-with-lease`

If there's no conflicts
- [ ] squash and merge
- [ ] delete branch

### CODE REVIEW

File: src/components/JWPlayerETP/Controls/BottomBar/index.js [hunk 1]
```diff
@@ -22,32 +26,72 @@ export const BottomBar = ({
...
-    </BarElementsContainer>
-  </BottomBarContainer>
-);
+  onSkipAd,
+  isAdPlaying,
+}) => {
+  const skipOffset = useCallback(() => {
+    if (seekPosition < 0) return false;
+
+    const offset = AD_SKIP_OFFSET - seekPosition;
```
Comment (JulienKode) [inline src/components/JWPlayerETP/Controls/BottomBar/index.js]: Can we move it to a custom hook file ?

File: src/components/JWPlayerETP/Controls/BottomBar/hooks.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { useMemo } from 'react';
+
+const useSkipOffset = ({ seekPosition, skipTime }) =>
```
Comment (MathieuDebit) [inline src/components/JWPlayerETP/Controls/BottomBar/hooks.js]: Forgot to rename 👀

File: src/components/JWPlayerETP/Controls/BottomBar/hooks.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  it('Should correctly compute skipTime when seekPosition is equal to 0', () => {
+    const { result } = renderHook(props => useSkipTime(props), {
+      initialProps: { seekPosition: 0, skipTime: offset },
+    });
+
+    expect(result.current).toEqual(offset + 1);
+  });
+
+  it('Should correctly compute skipTime when seekPosition is between 0 and AD_SKIP_OFFSET', () => {
```
Comment (MathieuDebit) [inline src/components/JWPlayerETP/Controls/BottomBar/hooks.test.js]: replace `AD_SKIP_OFFSET` by `skipTime` as constant does not exist anymore

File: src/components/JWPlayerETP/Controls/BottomBar/index.test.js [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
 import { mount } from 'enzyme';
 import { ThemeProvider } from 'emotion-theming';
 import theme from 'src/theme';
-import { BottomBar } from './index';
+import Spinner from 'src/elements/Spinner';
+import { BottomBar, SkipContainer, SkipLabel, SkipTimer, StyledSkipIcon } from './index';
 
 describe('Components|JWPlayerETP|Controls|BottomBar', () => {
   const Component = () => <div id="children-to-display" />;
+  const AD_SKIP_OFFSET = 6;
```
Comment (MathieuDebit) [inline src/components/JWPlayerETP/Controls/BottomBar/index.test.js]: useless now as it's `skipTime`

File: src/components/JWPlayerETP/Controls/BottomBar/index.test.js [hunk 5]
```diff
@@ -46,4 +54,141 @@ describe('Components|JWPlayerETP|Controls|BottomBar', () => {
...
+    const props = createDefaultProps({ isAdPlaying: true, seekPosition: AD_SKIP_OFFSET + 1.2345 });
+
+    const wrapper = mount(
+      <ThemeProvider theme={theme}>
+        <BottomBar {...props} />
+      </ThemeProvider>
+    ).find(BottomBar);
+
+    act(() => {
+      wrapper.find(SkipContainer).simulate('click');
```
Comment (thibaultboursier) [inline src/components/JWPlayerETP/Controls/BottomBar/index.test.js]: 👍

File: src/components/JWPlayerETP/Controls/BottomBar/index.js [hunk 6]
```diff
@@ -22,32 +26,66 @@ export const BottomBar = ({
...
+          <Volume volume={volume} mute={mute} onVolume={onVolume} onMute={onMute} />
+
+          <VerticalSeparator />
+
+          {isAdPlaying && skipEnabled && (
+            <SkipContainer onClick={onClickSkip} skipOffset={skipOffset}>
+              <SkipLabel skipOffset={skipOffset}>SKIP</SkipLabel>
+
+              <SkipTimerContainer skipOffset={skipOffset}>
+                {skipOffset ? (
```
Comment (JulienKode) [inline src/components/JWPlayerETP/Controls/BottomBar/index.js]: Can you add theses properties into the story to test it ?

File: src/components/JWPlayerETP/index.stories.js [hunk 7]
```diff
@@ -54,6 +54,8 @@ export const getBaseProps = (id, newProps) => {
     },
     LoginComponent: withVideoPlayerModal(LoginWithMarketing),
     recaptchaSiteKey: text('recaptchaSiteKey', '6LfvErIUAAAAABlpqACnxRiUhqhX4p14sPxx_sKf'),
+    skipEnabled: true,
```
Comment (JulienKode) [inline src/components/JWPlayerETP/index.stories.js]: Can we make that configurable ?

Comment (JulienKode) [reply src/components/JWPlayerETP/index.stories.js]: Should we add the same to `PlayerSkin` ?

File: src/components/JWPlayerETP/index.stories.js [hunk 8]
```diff
@@ -54,6 +54,8 @@ export const getBaseProps = (id, newProps) => {
...
+    skipTime: 6,
```
Comment (JulienKode) [inline src/components/JWPlayerETP/index.stories.js]: Can we add `onSkipAd` in the story

**Pull request merged**